float32 pose_x
float32 pose_y
float32 pose_theta

float32 x_final
float32 y_final
float32 theta_final
